### ISSUE TITLE
edit-config with delete not working 

### ISSUE THREAD

**Issue by meghna-doshi** [sysrepo/sysrepo#1846]
Hi,
I am using sysrepo (version 1.3.59) with Netopeer 2.1.1
When i give following input on netopeer2-cli, no deletion is reported by sysrepo to the subscribed application.

edit-config --target running --config --defop none
```
<model xmlns="urn:sysrepo:oven_model" xmlns:xc="urn:ietf:params:xml:ns:netconf:base:1.0" xc:operation="delete">
<name>lg</name>
</model>
```

At sysrepo i see following logs, looks like it is detecting no data change:

```
[INF]: LY: Resolving unresolved data nodes and their constraints...
[INF]: LY: All data nodes and constraints resolved.
[INF]: SR: Session 39 (user "root") created.
[INF]: LY: Resolving unresolved data nodes and their constraints...
[INF]: LY: All data nodes and constraints resolved.
[INF]: SR: Published event "rpc" with ID 8 priority 0 for 1 subscribers.
[INF]: SR: Processing "/ietf-netconf:edit-config" "rpc" event with ID 8 priority 0 (remaining 1 subscribers).
[INF]: NP: edit-config error-option "stop-on-error" not supported, rollback-on-error will be performed.
[INF]: SR: No datastore changes to apply.
[INF]: SR: Finished processing "rpc" event with ID 8 priority 0 (remaining 0 subscribers).
[INF]: SR: Event "rpc" with ID 8 priority 0 succeeded.
[INF]: NP: Session 2: thread 0 event new RPC.
```

yang module:

```
module oven_model {
    yang-version 1.1;
    namespace "urn:sysrepo:oven_model";
    prefix oven_model;
    revision 2018-01-19 {
        description "Initial revision.";
    }

    list model {
        description "Model container of the oven.";
        key "name";
        leaf name {
            description "identifier";
            type string;
        }
}
```

edit-config for addition done as follows:
```
edit-config --target running --config
<model xmlns="urn:sysrepo:oven_model">
    <name>lg</name>
    </model>

```

Can you please help here, why deletion is not working, though syntactically everything seems to be fine.

**Comment by michalvasko** [comment]
Hi,
the operation was being incorrectly replaced, fixed by ac9de6ecf62726794b6e5269da0e396bfac32c1c (version 1.4.17).

Regards,
Michal

**Comment by meghna-doshi** [comment]
Thanks Michal, it worked after taking latest code from devel branch.

Regards,
Meghna
